MGDAPI-3422 marin3r 0.8.0 -> 0.10.0

makslion opened the pull request:
# Issue link
[MGDAPI-3422](https://issues.redhat.com/browse/MGDAPI-3422)

# What
Update of the Marin3r 0.8.0 -> 0.10.0

# Verification steps
Verification would include a ratelimit test on the fresh install and after an upgrade. The proper way is to create your own index from this branch, but that might be time consuming, so if you want you can use my:
* Fresh install scenario: `quay.io/mvavilov/rhoam-index:1.19.0`
* Upgrade scenario: `quay.io/mvavilov/rhoam-index:1.20.0` will install the RHOAM from a maser (image created in the morning 24th February) and `quay.io/mvavilov/rhoam-index:1.21.0` will upgrade it to the image from this branch with a new Marin3r 
Here is the [guide](https://github.com/RHCloudServices/integreatly-help/blob/master/guides/olm/installing-rhmi-bundle-format.md) on how to do bundles and OLM installations 

## Fresh install 
1. Install operator and check the version of installed Marin3r operator - should be 0.10.0. Installation must be complete without any  problems
2. Run `./test/scripts/products/h22-validate-that-rate-limit-service-is-working-as-expected/test.sh` to ensure that ratelimiting works as expected 

## Upgrade 
1. Install operator with Marin3r 0.8.0 and then upgrade it to the version with Marin3r 0.10.0. The installation and upgrade must be completed without any problems. After the upgrade double-check that Marin3r is now at 0.10.0
2.  Run `./test/scripts/products/h22-validate-that-rate-limit-service-is-working-as-expected/test.sh` to ensure that ratelimiting works as expected

makslion commented:
/test rhoam-e2e

makslion commented:
/test rhoam-e2e

KevFan commented:
[Marin3r 0.9.0](https://github.com/3scale-ops/marin3r/releases/tag/v0.9.0) notes the following: 
![image](https://user-images.githubusercontent.com/24636860/156333023-7c25ac33-458e-49bd-93d3-5c31a5396b2e.png)

I think there would likely be v2 EnvoyConfigRevision is current production clusters so code probably needs to be added to delete these on upgrade :thinking:

makslion commented:
Hm, I've installed RHOAM using the image I've prepared for testing the upgrade scenario (the one with initial install) and `EnvoyConfigRevision` was at V3 already. 
![Screenshot from 2022-03-07 13-20-30](https://user-images.githubusercontent.com/37799801/157042158-83cf483d-c095-4dd2-a330-44712b5571aa.png)
Could it be because of [this](https://github.com/integr8ly/integreatly-operator/commit/cde0302f60b14dae9620cab93f47bf569ea7cde5#diff-eca908f8de77df9beb94a8c0dbc5b0a61b71487528ad7ac4c1bd790672a54162L22) PR that already addressed it? :thinking:

makslion commented:
rebased and 

/test rhoam-e2e

KevFan commented:
> Hm, I've installed RHOAM using the image I've prepared for testing the upgrade scenario (the one with initial install) and `EnvoyConfigRevision` was at V3 already. ![Screenshot from 2022-03-07 13-20-30](https://user-images.githubusercontent.com/37799801/157042158-83cf483d-c095-4dd2-a330-44712b5571aa.png) Could it be because of [this](https://github.com/integr8ly/integreatly-operator/commit/cde0302f60b14dae9620cab93f47bf569ea7cde5#diff-eca908f8de77df9beb94a8c0dbc5b0a61b71487528ad7ac4c1bd790672a54162L22) PR that already addressed it? thinking

Nope, I dont think the PR addresses that :thinking: It would be due to that PR and maybe possibly others that there would likely be `EnvoyConfigVersions` at V2 since it would upgrade from v2 to v3 :thinking: 

New installations from master would not see have `EnvoyConfigRevisions` at v2 as the PR been merged to use v3 so maybe the test should be upgrade of 1.17.0 (which does not contain the change to v3) to this PR :thinking:

makslion commented:
> > Hm, I've installed RHOAM using the image I've prepared for testing the upgrade scenario (the one with initial install) and `EnvoyConfigRevision` was at V3 already. ![Screenshot from 2022-03-07 13-20-30](https://user-images.githubusercontent.com/37799801/157042158-83cf483d-c095-4dd2-a330-44712b5571aa.png) Could it be because of [this](https://github.com/integr8ly/integreatly-operator/commit/cde0302f60b14dae9620cab93f47bf569ea7cde5#diff-eca908f8de77df9beb94a8c0dbc5b0a61b71487528ad7ac4c1bd790672a54162L22) PR that already addressed it? thinking
> 
> Nope, I dont think the PR addresses that thinking It would be due to that PR and maybe possibly others that there would likely be `EnvoyConfigVersions` at V2 since it would upgrade from v2 to v3 thinking
> 
> New installations from master would not see have `EnvoyConfigRevisions` at v2 as the PR been merged to use v3 so maybe the test should be upgrade of 1.17.0 (which does not contain the change to v3) to this PR thinking

Was able to confirm that upgrades from 1.17 to the image from this PR get stuck and the marin3r with "Awaiting Operator" and booth v2 and v3 revisions are present. I'll just do a sanity check with 1.17 -> master -> this PR and meanwhile will start working on the solving the issue

makslion commented:
Addressed the issue pointed out by @KevFan. Verification remains the same: I've updated the "fresh install" and "upgrade to" images and "upgrade from" is now at 1.17 (build from [this](https://github.com/integr8ly/integreatly-operator/pull/2452) PR). Versions that were shown in RHMI CR wasn't updated. 

/test rhoam-e2e

makslion commented:
/retest

makslion commented:
/test rhoam-e2e

makslion commented:
rebased against master to include the fix for the OO - was causing issues during the verification 

/test rhoam-e2e

bryankeane0 commented:
Marin3r bump is successful and the test passed on a fresh install. Both are also successful for upgrade scenario. Tested upgrade on OSD 4.9 and 1.17. Waiting to see if another upgrade test run is necessary on a more recent version of rhoam

bryankeane0 commented:
/lgtm

makslion commented:
/test images

makslion commented:
/retest

laurafitzgerald commented on pkg/products/marin3r/reconciler.go:
```diff
@@ -146,6 +147,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 
+	// Ensures that for an upgrade scenario we are removing V2 EnvoyConfigRevision before
+	// initializing an upgrade to marin3r 0.10.0
+	// TODO remove after the next release from here
```
Is there a JIRA for the removal?

laurafitzgerald commented:
/approve
